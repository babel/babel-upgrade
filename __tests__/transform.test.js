const fs = require('fs');
const path = require('path');
const pify = require('pify');
const transform = require('../src/transform');

async function readFile(configPath) {
  return (await pify(fs.readFile)(path.join(__dirname, configPath), 'utf8'));
}

describe('transform', () => {
  test('basic', async () => {
    const src = await readFile('../fixtures/transform/basic.js');
    expect(transform(src)).toMatchSnapshot();
  });

  test('function', async () => {
    const src = await readFile('../fixtures/transform/function.js');
    expect(transform(src)).toMatchSnapshot();
  });
});
