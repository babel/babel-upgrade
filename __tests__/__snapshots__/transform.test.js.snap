// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform basic 1`] = `
"module.exports = {
  plugins: [\\"@babel/plugin-transform-classes\\", \\"@babel/plugin-syntax-jsx\\"],
  presets: [\\"@babel/preset-env\\"]
};
"
`;

exports[`transform function 1`] = `
"function createBabelrc() {
  return {
    plugins: [
      \\"@babel/plugin-transform-classes\\",
      \\"@babel/plugin-syntax-jsx\\",
      \\"@babel/plugin-proposal-pipeline-operator\\"
    ],
    presets: [\\"@babel/preset-env\\", \\"@babel/preset-react\\"]
  };
}

module.exports = createBabelrc();
"
`;
